### DISCUSSION TITLE
Proposed changes to Unit 2, 3, 4, 6 projects.

### DISCUSSION THREAD

**Discussion by jdonwells** [TEALSK12/2nd-semester-introduction-to-computer-science#349] [Ideas]
A separate discussion was opened for changing the Unit 2 project to an expert system. https://github.com/TEALSK12/2nd-semester-introduction-to-computer-science/discussions/339

Remove lesson 2.06 because it isn't needed.

The Unit 3 project can be the Tic-Tac-Toe game from Unit 4. There are only 9 moves possible in a full game so a loop, while nice, isn't strictly required. Everything can be done with functions and no loops. 

Move Oregon Trail from Unit 3 to Unit 4 when loops are made available.

Move Zork from Unit 2 to Unit 6 when dictionaries are available. The Guess Who game seems to suffer without graphics anyway. Dictionaries will allow room contents to be a short string and translated into a longer description with a dictionary. Items in the bag can have short string identifications with longer descriptions in a dictionary for printing. Etc.

**Comment by georgiadoodle** [comment]
How did these proposed changes work out for you?

Also, did you reincorporate 2.06 later in the course?

**Reply by jdonwells** [reply]
They worked well in terms of projects the kids can handle given what they know. I created starter code for each project such that students were not writing hundreds of lines of code right away. 

The expert system introduces Neural Nets in a way they can easily understand. I give them an example with only 3 choices and ask for a different domain that has at least 6 choices.

For Tic-Tac-Toe I give them a 2x2 version. They need to add about a dozen lines of code and modify a dozen other lines.

Oregon Trail has starter code that needs about 80 lines of code added.

Zork has starter code that needs about 100 lines of code added.

Pokemon also needs about 100 lines of code.

I feel like it is not overwhelming to the students. Plus they need to develop the skill of reading code. 

2.06 just isn't needed. It was added to the curriculum to be able to make Zork possible as a Unit 2 project. Moving Zork out of there frees them from the tyranny of if elif that marches off the page. Doing Zork when we have dictionaries makes it possible to do with a data driven style. The starter code includes an abstraction layer for the map.

**Comment by jdonwells** [comment]
Starter code for these projects can be found here https://replit.com/@DonWells
